
From: Rusty Russell <rusty@rustcorp.com.au>

From: Pavel Roskin <proski@gnu.org>

If you use both module_param (new) and MODULE_PARM (obsolete) in a module,
only the second gets recognised.  Warn.


---

 25-akpm/kernel/module.c |    4 ++++
 1 files changed, 4 insertions(+)

diff -puN kernel/module.c~warn-if-module_param-and-module_parm-mixed kernel/module.c
--- 25/kernel/module.c~warn-if-module_param-and-module_parm-mixed	2004-04-17 21:39:55.897624016 -0700
+++ 25-akpm/kernel/module.c	2004-04-17 21:39:55.901623408 -0700
@@ -1543,6 +1543,10 @@ static struct module *load_module(void _
 				      / sizeof(struct obsolete_modparm),
 				      sechdrs, symindex,
 				      (char *)sechdrs[strindex].sh_addr);
+		if (setupindex)
+			printk(KERN_WARNING "%s: Ignoring new-style "
+			       "parameters in presence of obsolete ones\n",
+			       mod->name);
 	} else {
 		/* Size of section 0 is 0, so this works well if no params */
 		err = parse_args(mod->name, mod->args,

_
